test/UnitTests/CheckoutControllerTest.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.Http.Core.Collections;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.DependencyInjection.Fallback;
+using Xunit;
+using MusicStore.Controllers;
+using MusicStore.Models;
+
+namespace UnitTests
+{
+    public class CheckoutControllerTest
+    {
+        IServiceProvider _serviceProvider;
+
+        public CheckoutControllerTest()
+        {
+            var services = new ServiceCollection();
+
+            services.AddEntityFramework()
+                      .AddInMemoryStore()
+                      .AddDbContext<MusicStoreContext>();
+
+            _serviceProvider = services.BuildServiceProvider();
```

Comment:
this is how we generate DbContexts for tests, but you should ensure that they are disposed by doing something like this

```
            var options = new DbContextOptions()
                .UseModel(model);
            options.UseInMemoryStore();

            using (var context = new DbContext(ServiceProvider, options))
            {
            }
```